### PR TITLE
Record attch/FLCRM6842: Add name to attachment value item

### PR DESCRIPTION
**Pull request by Jirles**
# What
- Add's a `name` value for the `AttachmentItemValue` 
- Also overwrites some functions to better handle how attachments are different from other media values, i.e. no captions, only filename
- Refactors tests to pass again

# Why
Issue is here: [FLCRM-6842](https://fulcrumapp.atlassian.net/browse/FLCRM-6842)

Blocks [FLCRM-6798](https://fulcrumapp.atlassian.net/browse/FLCRM-6798) 

# Testing
_Unit_
- Refactored tests

_Manual_
- If you really want, you can also manually test this with this draft PR: https://github.com/fulcrumapp/fulcrum/pull/5040
- Just update the `fulcrum-core` version in the `package.json` file with this value: `"fulcrumapp/fulcrum-core#record-attch\/FLCRM-6842-add-filename-to-media-item-value"`

### CODE REVIEW

**Review by jahndis — CHANGES_REQUESTED** [review]
It looks like `yarn build` needs to be run again to generate the dist for attachment-value.js and map

**Review by jahndis — APPROVED** [review]
Let me know when you're ready to merge this and I can help with publishing.

**Comment by Jirles** [comment]
> Let me know when you're ready to merge this and I can help with publishing.

Thanks! I'll let you know. I'm checking that it works as I think it will right now but hopefully soon. Also I feel like I should get one more approval since we normally request two before a merge...

File: src/values/attachment-item-value.js
Comment (shoesCodeFor) [inline src/values/attachment-item-value.js:7]: Is this the filename?

File: test/elements/attachment-element.js
Comment (shoesCodeFor) [inline test/elements/attachment-element.js:36]: I guess this answers my question.  I would prefer `filename` or `file_name` because it seems more descriptive IMHO.

**Review by shoesCodeFor — CHANGES_REQUESTED** [review]

File: src/values/attachment-item-value.js
Comment (Jirles) [reply src/values/attachment-item-value.js:7]: Yes. I _think_ we're moving to using `name` instead of `filename` so I went with name here.

**Comment by shoesCodeFor** [comment]
I'm not opposed to merging this as is, I just wanted to make a suggestion on the name.

File: src/values/attachment-item-value.js
Comment (shoesCodeFor) [reply src/values/attachment-item-value.js:7]: I'm seeing this in some of the React components: `attachment.filename`
But you're right it says `name` in the attachments service.  I'm going to pull this convo into Slack

**Review by shoesCodeFor — APPROVED** [review]

**Pull request merged**
